### DISCUSSION TITLE
"not" operator in filters

### DISCUSSION THREAD

**Discussion by eterX** [ransome1/sleek#496] [Ideas]
Hi everyone,

What if repeated clicking in some tag, project or context, turns that tag negative in the filter. i.e. show only tasks that DO NOT have it. Today, click selects or deselects it. This feature would add a new state to the tag, see PRJ2 in this capture:

![image](https://github.com/ransome1/sleek/assets/6190461/801afdd5-df24-4ea2-ba0d-e11feb1fdd30)

Thanks

PS: Sleek rocks!

**Comment by ransome1** [comment]
I'll see what I can do once I finish the currently ongoing rewriting process.

**Comment by ransome1** [comment]
In the latest developer preview of sleek (it is being rewritten at the moment), this enhanced way of filtering has been added. In order to exclude filters, you need to press and hold Ctrl (or CMD on Mac) and you'll see the filter buttons change their appearance. If you now select one, this filter will be added to the exclude filter list. In your case todos with the project `PRJ2` will not appear any longer in the todo table. 

Please feel free to check it and give me some feedback on it: https://github.com/ransome1/sleek/releases/tag/v2.0.0-dev3

**Comment by eterX** [comment]
Hi Randome1,

Thank you very much guys for this new feature! (long awaited, btw) :D

Tested it (v2.0.0-dev3, Mac-ARM and Linux-Appimage) in any ways I could thought of and works great. PLUS, looks gorgeous too. 👏🏽 

The only tiny observation I'm able to contribute, just for the sake of making myself slightly useful is: CTRL+click on a selected tag doesn't work as intuition would dictate. I.e. in this state:

![image](https://github.com/ransome1/sleek/assets/6190461/19eaf6f8-d971-445a-a92b-1668eb233146)

CTRL+clicking on `PRJ1` will deselect it, instead of turning it directly to `¬PRJ1` ...nothing serious...  :)

Thanks again. Sleek rocks (even more!)

**Reply by ransome1** [reply]
Thanks for your feedback, I'll look into it

**Reply by ransome1** [reply]
Should be fixed in Dev4, coming soon.
